nixos/modules/services/networking/adguardhome.nix

```diff
@@ -0,0 +1,97 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.adguardhome;
+
+  args = concatStringsSep " " ([
+    "--no-check-update"
+    "--config ${cfg.confFile}"
+    "--work-dir ${cfg.workDir}"
+    "--host ${cfg.host}"
+    "--port ${toString cfg.port}"
+  ] ++ optional (cfg.logFile != null) "--logfile ${cfg.logFile}"
+  ++ optional (cfg.pidFile != null) "--pidfile ${cfg.pidFile}"
+  ++ optional cfg.verbose "--verbose");
+
+in
+{
+  options.services.adguardhome = with types; {
+    enable = mkEnableOption "AdGuard Home network-wide ad blocker";
+
+    confFile = mkOption {
+      default = "/etc/AdGuardHome/AdGuardHome.yaml";
+      type = path;
+      description = ''
+        Path to the config file.
+      '';
+    };
+
+    workDir = mkOption {
```

Comment:
Usually we call this `stateDir`, depending on what it is for...

Reply:
This is where AdGuard Home stores filter lists / it's internal database. I was just copying AdGuard Home's terminology for `workDir` (which I saw used in a few other modules). But `stateDir` is more appropriate, I'll change it (in the morning 😴).